packages/test/src/Happy/Test/Shell.hs

```diff
@@ -0,0 +1,55 @@
+module Happy.Test.Shell where
+
+import System.Directory
+import System.Process
+import System.Exit
+import Control.Monad
+import Control.Monad.Trans.Maybe
+import Control.Monad.IO.Class
+import Control.Applicative
+
+type TypedShell a = MaybeT IO a
```

Comment:
This is all a bit ad-hoc and could probably take inspiration in e.g. [`shelly`](https://hackage.haskell.org/package/shelly) or [`turtle`](https://hackage.haskell.org/package/turtle). Then again, if it works it's probably good enough. Although we already had experiences that were hard to debug...

Reply:
Yeah as long as this doesn't need to change too often I don't care too much.